examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,647 @@
+             * doesn't support WPA2, these mode can be enabled by commenting below line */
+	        .threshold = {
+                .authmode = static_cast<wifi_auth_mode_t>(request.security_type),
+            },
+
+            .pmf_cfg = {
+                .capable = true,
+                .required = false
+            },
+        },
+    };
+        memcpy(wifi_config.sta.ssid, request.ssid.bytes,
+               std::min(sizeof(wifi_config.sta.ssid), static_cast<size_t>(request.ssid.size)));
+        memcpy(wifi_config.sta.password, request.secret.bytes,
+               std::min(sizeof(wifi_config.sta.password), static_cast<size_t>(request.secret.size)));
+        PW_ESP_TRY(wifi_connect_sta(&wifi_config, &response));
+        return pw::OkStatus();
+    }
+
+    pw::Status Disconnect(ServerContext &, const chip_rpc_Empty & request, chip_rpc_Empty & response)
+    {
+        PW_ESP_TRY(esp_wifi_disconnect());
+        return pw::OkStatus();
+    }
+};
+
+} // namespace rpc
+} // namespace chip
+
+namespace {
+
+using std::byte;
+
+constexpr size_t kRpcStackSizeBytes         = (4 * 1024);
+constexpr uint8_t kRpcTaskPriority          = 5;
+constexpr size_t kScanResultsStackSizeBytes = (6 * 1024);
+constexpr uint8_t kScanResultsTaskPriority  = 3;
+constexpr size_t kTestStackSizeBytes        = (8 * 1024);
+constexpr uint8_t kTestTaskPriority         = 5;
+
+SemaphoreHandle_t uart_mutex;
+
+TaskHandle_t rpcTaskHandle;
+TaskHandle_t scanResultsTaskHandle;
+TaskHandle_t testTaskHandle;
+
+chip::rpc::GDMWifiBase gdm_wifi_service;
+
+void RegisterServices(pw::rpc::Server & server)
+{
+    server.RegisterService(gdm_wifi_service);
+}
+
+void RunRpcService(void *)
+{
+    ::chip::rpc::Start(RegisterServices, &::chip::rpc::logger_mutex);
+}
+
+} // namespace
+
+extern "C" void app_main()
+{
+    PigweedLogger::init();
+    uartInitialised = true;
+    // Initialize NVS
+    esp_err_t ret = nvs_flash_init();
+    if (ret == ESP_ERR_NVS_NO_FREE_PAGES || ret == ESP_ERR_NVS_NEW_VERSION_FOUND)
+    {
+        ESP_ERROR_CHECK(nvs_flash_erase());
+        ret = nvs_flash_init();
+    }
+    ESP_ERROR_CHECK(ret);
+
+    wifi_init_semaphore = xSemaphoreCreateBinary();
+    ESP_LOGI(TAG, "wifi_init_sta: %s", pw_StatusString(wifi_init_sta()));
+
+    ESP_LOGI(TAG, "----------- chip-esp32-ipv6-example starting -----------");
+
+    uart_mutex            = xSemaphoreCreateMutex();
+    scanresults_semaphore = xSemaphoreCreateBinary();
+    xTaskCreate(RunRpcService, "RPC", kRpcStackSizeBytes / sizeof(StackType_t), nullptr, kRpcTaskPriority, &rpcTaskHandle);
+
+    xTaskCreate(ReturnScanResults, "ScanResults", kScanResultsStackSizeBytes / sizeof(StackType_t), nullptr,
+                kScanResultsTaskPriority, &scanResultsTaskHandle);
+    xTaskCreate(&UdpReceiver, "TestTask", kTestStackSizeBytes / sizeof(StackType_t), nullptr, kTestTaskPriority, &testTaskHandle);
```

Comment:
`&UdpReceiver` -> `UdpReceiver` for consistency with above?

Reply:
done